File: BarberMe/BarberMe.Database/Models/ShopWorkingHours.cs
namespace BarberMe.Database.Models
{
    public class ShopWorkingHours
    {
        public int ShopWorkingHoursId { get; set; }

        public int DayOfWeek { get; set; }

        public TimeSpan StartTime { get; set; }

        public TimeSpan EndTime { get; set; }

        public bool IsWorking { get; set; }
    }
}

File: barberme_mobile/lib/models/shop_working_hours.dart
class ShopWorkingHours {
  final int id;
  final int dayOfWeek;
  final String startTime;
  final String endTime;
  final bool isWorking;

  ShopWorkingHours({
    required this.id,
    required this.dayOfWeek,
    required this.startTime,
    required this.endTime,
    required this.isWorking,
  });

  factory ShopWorkingHours.fromJson(Map<String, dynamic> json) {
    return ShopWorkingHours(
      id: json['id'] as int,
      dayOfWeek: json['dayOfWeek'] as int,
      startTime: json['startTime']?.toString() ?? '',
      endTime: json['endTime']?.toString() ?? '',
      isWorking: json['isWorking'] as bool? ?? false,
    );
  }
}